decision/value_functions/ann_value_function.py

```diff
@@ -0,0 +1,62 @@
+from .abstract import *
+from tensorflow import keras
+import numpy as np
+
+
+class ANNValueFunction(ValueFunction):
+    def __init__(self, network_structure: [int], **kwargs):
+        super().__init__(**kwargs)
+        self.network_structure = network_structure
+        self.model = keras.Sequential()
+
+    def setup(self, state: classes.State):
+        (
+            number_of_locations_indicators,
+            number_of_state_features,
+        ) = self.get_number_of_location_indicators_and_state_features(state)
+        # First layer needs input size as argument
+        nodes_in_first_layer, *nodes_in_rest_of_layers = self.network_structure
+        self.model.add(
+            keras.layers.Dense(
+                nodes_in_first_layer,
+                input_dim=(number_of_locations_indicators + number_of_state_features),
+            )
+        )
+        for layer in nodes_in_rest_of_layers:
+            self.model.add(keras.layers.Dense(layer))
+        # The last layer needs to have a single value function output
+        self.model.add(keras.layers.Dense(1))
+        self.model.compile(
+            loss="mean_squared_error", optimizer="SGD", metrics=["accuracy"]
+        )
+        super(ANNValueFunction, self).setup(state)
+
+    def estimate_value(
+        self,
+        state: classes.State,
+        vehicle: classes.Vehicle,
+        time: int,
+        state_features=None,
+    ):
+        if not state_features:
+            state_features = self.get_state_features(state, vehicle, time)
+
+        return float(
+            self.model.predict(
+                np.array(state_features).reshape(1, (len(state_features)))
+            )[0][0]
```

Comment:
Is i remember from AI-prog, ```model(input).numpy()``` is much faster then ```model.predict(inputs)``` as the first one returns a tensor. Since estimate value is being used during the roll outs, I think this can enhance computational time.